File: lib/src/domain/khipu_event.dart
class KhipuEvent {
  final String name;
  final String timestamp;
  final String type;

  KhipuEvent({
    this.name = '',
    this.timestamp = '',
    this.type = '',
  });

  factory KhipuEvent.fromJson(Map<String, dynamic> json) {
    return KhipuEvent(
      name: json['name'] ?? '',
      timestamp: json['timestamp'] ?? '',
      type: json['type'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'timestamp': timestamp,
      'type': type,
    };
  }

  @override
  String toString() {
    return '{name: $name, timestamp: $timestamp, type: $type}';
  }
}

File: ios/Classes/Models/KhipuEvent.swift
struct KhipuEvent: Codable {
  let name: String
  let timestamp: String
  let type: String

  init(
    name: String = "",
    timestamp: String = "",
    type: String = ""
  ) {
    self.name = name
    self.timestamp = timestamp
    self.type = type
  }
}
